### DISCUSSION TITLE
Rotate door in Portainer

### DISCUSSION THREAD

**Discussion by Bosswaffle** [mrlt8/docker-wyze-bridge#474] [Q&A]
Howdy, I set this up in Portainer per the [wiki](https://github.com/mrlt8/docker-wyze-bridge/wiki/Portainer), and added "ROTATE_DOOR" as an environment variable, but by doorbell camera stream isn't rotated.

Is this because I'm using 'docker-compose.sample.yml' instead of my own .yml file?

**Comment by mrlt8** [comment]
I believe it should work with the sample yaml. Does restarting the container get it to read the new env?

**Reply by Bosswaffle** [reply]
Unfortunately neither restarting the stack or the container seems to have any effect.  
![image](https://user-images.githubusercontent.com/29218546/178900412-cb44d5d7-4769-4b02-a44c-d09593f44e3f.png)
Here's a screenshot of most of my environment variables

**Reply by mrlt8** [reply]
hmm, that is strange. Is your doorbell showing up as a "doorbell" in the logs?

Changes do seem to reflect in the container for me with: sample config -> containers -> <my_container> -> duplicate/edit -> ENV -> <changes> -> "Deploy the container"

**Reply by Bosswaffle** [reply]
Here's a sample from the log, it does appear as a doorbell:
`2022/07/14 15:00:24 [WyzeBridge] 🎉 Connecting to WyzeCam Doorbell - Front Doorbell on 192.168.20.21 (1/3)`

I did add my ENV variables to the "stack" and they don't appear to make it to the container's ENV variables. When I add the 'Rotate_Door' variable to the container's ENV directly it does work.

It is obvious that I'm a docker noob?

**Reply by latetedemelon** [reply]
Not sure if this got solved but I'm having the same issue in Portainer.  I tried rotating by naming the camera directly and rotating all doorbells with no luck.

**Reply by Bosswaffle** [reply]
I had to edit the container itself and add the "Rotate_Door" ENV variable there. **This does not persist when redploying the stack.** It still seems to me that the stack's ENV variables are only pulled from the docker-compose.sample.yml file. 

I haven't tried any of the newer features, but I think that it would behave the same for: ON_DEMAND, ENABLE_AUDIO, FILTER_MODELS, etc.

**Reply by Bosswaffle** [reply]
I just tested by adding "FILTER_MODELS=V2" to the stack's listy of ENV variables, and redeployed the stack, none of the ENV variables other than WYZE_EMAIL and WYZE_PASSWORD made it to the container. I did have to add "ROATE_DOOR=true" to the container again.
